### DISCUSSION TITLE
Detect SSH login attempt on a single port only

### DISCUSSION THREAD

**Discussion by Deewde** [fail2ban/fail2ban#3471] [Q&A]
Is there a way to detect a failed login attempt on the configured port **_only_** and take action normally? Consider this:

> [sshd]
> port = 7777

In auth.log this should be ignored because the attempt is made on a different port:

> Mar  4 22:10:28 vmi725339 sshd[1320766]: Invalid user anang from 82.65.173.65 port 48768
> Mar  4 22:10:28 vmi725339 sshd[1320766]: pam_unix(sshd:auth): check pass; user unknown
> Mar  4 22:10:28 vmi725339 sshd[1320766]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=82.65.173.65
> Mar  4 22:10:29 vmi725339 sshd[1320766]: Failed password for invalid user anang from 82.65.173.65 port 48768 ssh2
> Mar  4 22:10:30 vmi725339 sshd[1320766]: Received disconnect from 82.65.173.65 port 48768:11: Bye Bye [preauth]
> Mar  4 22:10:30 vmi725339 sshd[1320766]: Disconnected from invalid user anang 82.65.173.65 port 48768 [preauth]

but the below should be considered regarding a regular action

> Mar  4 22:10:28 vmi725339 sshd[1320766]: Invalid user anang from 82.65.173.65 port 7777
> Mar  4 22:10:28 vmi725339 sshd[1320766]: pam_unix(sshd:auth): check pass; user unknown
> Mar  4 22:10:28 vmi725339 sshd[1320766]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=82.65.173.65
> Mar  4 22:10:29 vmi725339 sshd[1320766]: Failed password for invalid user anang from 82.65.173.65 port 48768 ssh2
> Mar  4 22:10:30 vmi725339 sshd[1320766]: Received disconnect from 82.65.173.65 port 48768:11: Bye Bye [preauth]
> Mar  4 22:10:30 vmi725339 sshd[1320766]: Disconnected from invalid user anang 82.65.173.65 port 48768 [preauth]`

**Comment by sebres** [comment]
I don't think the approach is faciable - the port you see in the messages like `from IP port 48768` is almost always a remote port (e.g. 48768 is a dynamic port of other side)… where 7777 surely means a local port (one of the listener of sshd)...
I guess you'd need to consider some other value (e.g. service name or whatever else is different).
For systemd journal you can try to output both excerpt in JSON, find the difference and filter it by `journalmatch` specifying per jail.
